apps/dashboard/src/hooks/use-inbox-integration-workflow-updater.ts

```diff
@@ -0,0 +1,218 @@
+
+      const runNext = async (): Promise<void> => {
+        const currentIndex = index++;
+        if (currentIndex >= items.length) return;
+
+        try {
+          const result = await worker(items[currentIndex]);
+          results.push(result);
+        } finally {
+          await runNext();
+        }
+      };
+
+      const runners = Array.from({ length: Math.min(maxConcurrency, items.length) }, () => runNext());
+      await Promise.all(runners);
+
+      return results;
+    },
+    [concurrency]
+  );
+
+  const initializeProcessingState = useCallback((workflows: WorkflowListResponseDto[]): WorkflowProcessingState => {
+    const slugToWorkflow = new Map<WorkflowSlug, WorkflowListResponseDto>(
+      workflows.map((workflow) => [workflow.slug, workflow])
+    );
+
+    return {
+      slugToWorkflow,
+      slugs: Array.from(slugToWorkflow.keys()),
+      deactivated: new Set<WorkflowSlug>(),
+      reactivated: new Set<WorkflowSlug>(),
+      deactivateErrors: new Map<WorkflowSlug, Error>(),
+      reactivateErrors: new Map<WorkflowSlug, Error>(),
+      workflowCache: new Map<WorkflowSlug, WorkflowResponseDto>(),
+    };
+  }, []);
+
+  const deactivateWorkflow = useCallback(
+    async (slug: WorkflowSlug, environment: NonNullable<typeof currentEnvironment>, state: WorkflowProcessingState) => {
+      try {
+        await patchWorkflow({
+          environment,
+          workflowSlug: slug,
+          workflow: { active: false },
+        });
+
+        state.deactivated.add(slug);
+      } catch (error) {
+        state.deactivateErrors.set(slug, error as Error);
+      }
+    },
+    []
+  );
+
+  const reactivateWorkflow = useCallback(
+    async (slug: WorkflowSlug, environment: NonNullable<typeof currentEnvironment>, state: WorkflowProcessingState) => {
+      // Skip reactivation for workflows that failed to deactivate to avoid inconsistent states
+      if (state.deactivateErrors.has(slug)) {
+        return;
+      }
+
+      try {
+        await patchWorkflow({
+          environment,
+          workflowSlug: slug,
+          workflow: { active: true },
+        });
+
+        state.reactivated.add(slug);
+      } catch (error) {
+        state.reactivateErrors.set(slug, error as Error);
+      }
+    },
+    []
+  );
+
+  const refreshWorkflow = useCallback(
+    async (
+      slug: WorkflowSlug,
+      environment: NonNullable<typeof currentEnvironment>,
+      state: WorkflowProcessingState,
+      forceActive?: boolean
+    ) => {
+      try {
+        let workflowData = state.workflowCache.get(slug);
+        if (!workflowData) {
+          workflowData = await getWorkflow({ environment, workflowSlug: slug });
+          state.workflowCache.set(slug, workflowData);
+        }
+
+        await patchWorkflow({
+          environment,
+          workflowSlug: slug,
+          workflow: {
+            active: forceActive !== undefined ? forceActive : workflowData.active,
+          },
+        });
```

Comment:
@iampearceman but why do we change the active state? What if the user had a workflow that was disabled? We shouldn't enable it in this flow...